src/GLModel.ts

```diff
@@ -175,82 +175,80 @@ export class GLModel {
 
-            var offset = geoGroup.vertices * 3;
-
-            geoGroup.vertices++;
-            vertexArray[offset] = atom.x + points[j][0];
-            vertexArray[offset + 1] = atom.y + points[j][1];
-            vertexArray[offset + 2] = atom.z + points[j][2];
-            colorArray[offset] = c.r;
-            colorArray[offset + 1] = c.g;
-            colorArray[offset + 2] = c.b;
+            vertices[offset] = atom.x + points[j][0];
+            vertices[offset + 1] = atom.y + points[j][1];
+            vertices[offset + 2] = atom.z + points[j][2];
+            colors[offset] = c.r;
+            colors[offset + 1] = c.g;
+            colors[offset + 2] = c.b;
 
             if (clickable) {
-                var point = new Vector3(points[j][0], points[j][1], points[j][2]);
+                const point = new Vector3(points[j][0], points[j][1], points[j][2]);
 
                 //decrease cross size for selection to prevent misselection from atom overlap
                 point.multiplyScalar(0.1);
                 point.set(point.x + atom.x, point.y + atom.y, point.z + atom.z);
                 atom.intersectionShape.line.push(point);
             }
-
         }
 
+        const vertexArray = geoGroup.vertexArray;
+        const colorArray = geoGroup.colorArray;
+        const startOffset = geoGroup.vertices * 3;
+        const endOffset = startOffset + vertices.length;
+
+        vertexArray.set(vertices, startOffset);
+        colorArray.set(colors, startOffset);
+
+        geoGroup.vertices += vertices.length / 3;
     };
 
-    private getGoodCross(atom:AtomSpec, atom2:AtomSpec, p1, dir) {
-        // get vector 2 different neighboring atom
-        //find most divergent neighbor
-        var bestv = null;
-        var bestlen = -1;
-        for (var j = 0, n = atom.bonds.length; j < n; j++) {
-            if (atom.bonds[j] != atom2.index) {
-                let j2 = atom.bonds[j];
-                let atom3 = this.atoms[j2];
-                let p3 = new Vector3(atom3.x, atom3.y, atom3.z);
-
-                let dir2 = p3.clone();
-                dir2.sub(p1);
 
-                let v = dir2.clone();
-                v.cross(dir);
-                var l = v.lengthSq();
+    private getGoodCross(atom: AtomSpec, atom2: AtomSpec, p1: Vector3, dir: Vector3): Vector3 | null {
+        // Get vector to different neighboring atom and find most divergent neighbor
+        let bestv: Vector3 | null = null;
+        let bestlen = -1;
+        for (const bond of atom.bonds) {
+            if (bond !== atom2.index) {
+                const [j2, atom3] = [bond, this.atoms[bond]];
+                const p3 = new Vector3(atom3.x, atom3.y, atom3.z);
+                const dir2 = p3.clone().sub(p1);
+                const v = dir2.clone().cross(dir);
+                const l = v.lengthSq();
                 if (l > bestlen) {
                     bestlen = l;
                     bestv = v;
-                    if (bestlen > 0.1) {
```

Comment:
You've removed the short circuiting of the loop so now it will run longer.  Why do you think this is an improvement?